src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/VariantArray.cs

```diff
@@ -49,11 +50,23 @@ internal static class VariantArray
         // (guarenteed less than 28, in practice 0-2)
         private static readonly List<Type> s_generatedTypes = new List<Type>(0);
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray1))]
```

Comment:
(nit) could be
```suggestion
        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicFields, typeof(VariantArray1))]
```

But since the type only contains fields, it isn't super important.